test/jdk/javax/swing/JMenuItem/8031573/bug8031573.java

```diff
@@ -20,45 +20,110 @@
+            + "2. Open the Menu.\n"
+            + "3. Check that the icon on the JCheckBoxMenuItem is smooth.\n"
+            + "   If so, press PASS, else press FAIL.\n";
 
-            SwingUtilities.invokeAndWait(new Runnable() {
+    public static void main(String args[]) throws Exception {
+        UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+        try {
+            SwingUtilities.invokeAndWait(() -> createTestGUI());
 
-                @Override
-                public void run() {
-                    JMenuBar bar = new JMenuBar();
-                    JMenu menu = new JMenu("Menu");
-                    JCheckBoxMenuItem checkBoxMenuItem
-                            = new JCheckBoxMenuItem("JCheckBoxMenuItem");
-                    checkBoxMenuItem.setSelected(true);
-                    menu.add(checkBoxMenuItem);
-                    bar.add(menu);
-                    setJMenuBar(bar);
+            if (!latch.await(5, TimeUnit.MINUTES)) {
+                throw new RuntimeException("Test has timed out!");
+            }
+            if (!passed) {
+                throw new RuntimeException("Test failed!");
+            }
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
                 }
             });
-        } catch (Exception e) {
-            throw new RuntimeException(e);
         }
     }
+
+    private static void createTestGUI() {
+        frame = new JFrame("bug8031573");
+        JMenuBar bar = new JMenuBar();
+        JMenu menu = new JMenu("Menu");
+        JCheckBoxMenuItem checkBoxMenuItem = new JCheckBoxMenuItem("JCheckBoxMenuItem");
+        checkBoxMenuItem.setSelected(true);
+        menu.add(checkBoxMenuItem);
+        bar.add(menu);
+        frame.setJMenuBar(bar);
+
+        JPanel panel = new JPanel(new BorderLayout());
+        JTextComponent textComponent = new JTextArea(INSTRUCTIONS);
+        textComponent.setEditable(false);
+        panel.add(textComponent, BorderLayout.CENTER);
+
+        JPanel buttonsPanel = new JPanel(new FlowLayout());
+        JButton passButton = new JButton("Pass");
+        passButton.addActionListener((e) -> {
+            System.out.println("Test passed!");
+            passed = true;
+            latch.countDown();
+        });
+        JButton failsButton = new JButton("Fail");
+        failsButton.addActionListener((e) -> {
+            passed = false;
+            latch.countDown();
+        });
+
+        buttonsPanel.add(passButton);
+        buttonsPanel.add(failsButton);
+        panel.add(buttonsPanel, BorderLayout.SOUTH);
+
+        frame.getContentPane().add(panel);
+
+        frame.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent e) {
+                latch.countDown();
+            }
+        });
+        frame.setSize(760, 250);
+        frame.pack();
```

Comment:
`setSize` is redundant when you call `pack`. The latter will also set size of the frame based on the preferred size determined by the layout manager.